src/storage/txn/scheduler.rs

```diff
@@ -207,6 +207,10 @@ impl<E: Engine> Scheduler<E> {
         let tctx = {
             let cmd = task.cmd();
             let lock = self.gen_lock(cmd);
+            // Write command should acquire write lock.
+            if !cmd.readonly() && !lock.is_write_lock() {
```

Comment:
have we met this problem before?

Reply:
Yes, i remember the first version of resolve lock forget to acquire latch and caused data corruption.

Reply:
I think there may be a more elegant way but I cannot come up with an idea. LGTM.

Reply:
I also forget to acquire latch for pessimistic locks at the beginning.